great_expectations/agent/event_handler.py

```diff
@@ -26,28 +16,19 @@ class EventHandler:
     def __init__(self, context: CloudDataContext) -> None:
         self._context = context
 
-    def handle_event(self, event_context: EventContext) -> EventHandlerResult:
-        """Pass event to the correct handler."""
-        if isinstance(event_context.event, RunDataAssistantEvent):
-            return self._handle_run_data_assistant(event_context)
-        elif isinstance(event_context.event, RunCheckpointEvent):
-            return self._handle_run_checkpoint(event_context)
+    def handle_event(self, event: Event, id: str) -> ActionResult:
+        """Transform an Event into an ActionResult."""
+
+        if isinstance(event, RunOnboardingDataAssistantEvent):
+            action = RunOnboardingDataAssistantAction(context=self._context)
+        elif isinstance(event, RunCheckpointEvent):
+            raise NotImplementedError
```

Comment:
I would seriously consider using `singledispatch` for this.
It's the perfect use-case where we want to handle something differently according to the type of one of the arguments.

https://docs.python.org/3/library/functools.html#functools.singledispatch
https://www.educative.io/answers/what-is-singledispatch-descriptor-in-functools-module-in-python

Reply:
yeah, that's a valid solution, but in my experience people have a more difficult time reading and understanding `singledispatch` methods. Let's go with the less magical solution, and if in the future this gets unwieldy we can implement this in a different way.